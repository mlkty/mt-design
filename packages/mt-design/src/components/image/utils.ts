
export const isValidImage = (src: string) => {
    return new Promise(resolve => {
        const image = new Image();
        image.onload = () => resolve(true);
        image.onerror = () => resolve(false);
        image.src = src;
    });
};
