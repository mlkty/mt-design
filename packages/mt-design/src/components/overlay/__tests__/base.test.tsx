/* eslint-disable max-len */
import {Overlay} from '..';
import {customRender, sleep} from '../../../../tests/utils';
import mountTest from '../../../../tests/mountTest';

describe('Overlay', () => {
    mountTest(Overlay);

    it('render snapshot when visible correctly', () => {
        const {container} = customRender(<Overlay visible />);
        expect(container).toMatchSnapshot();
    });

    it('render classnames correctly througn invisible to visible', async () => {
        const {container, rerender} = customRender(<Overlay />);
        rerender(<Overlay visible />);
        expect(container.firstChild).toHaveClass('mt-overlay-fade-enter mt-overlay-fade-enter-active');
        await sleep(350);
        expect(container.firstChild).toHaveClass('mt-overlay-fade-enter-done');
    });

    it('render classnames correctly througn visible to invisible', async () => {
        const {container, rerender} = customRender(<Overlay visible />);
        expect(container.firstChild).toHaveClass('mt-overlay-fade-appear mt-overlay-fade-appear-active');
        rerender(<Overlay />);
        expect(container.firstChild).toHaveClass('mt-overlay-fade-exit mt-overlay-fade-exit-active');
        await sleep(350);
        expect(container).toBeEmptyDOMElement();
    });

    it('initialize have dom', () => {
        const {container} = customRender(<Overlay forceRender />);
        expect(container.firstChild).toHaveClass('mt-overlay');
    });

    it('unmounted keep dom', () => {
        const {container, rerender} = customRender(<Overlay visible />);
        rerender(<Overlay destroyOnHide={false} />);
        expect(container.children.length).toBe(1);
    });
});
