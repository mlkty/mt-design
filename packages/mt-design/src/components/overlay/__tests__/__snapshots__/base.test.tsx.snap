// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay render snapshot when visible correctly 1`] = `
<div>
  <div
    class="mt-overlay mt-overlay--force mt-overlay-fade-appear mt-overlay-fade-appear-active"
    style="--duration: 0.3s; --z-index: 10;"
  />
</div>
`;
